Sources/PackageLoading/TargetSourcesBuilder.swift

```diff
@@ -247,6 +248,34 @@ public struct TargetSourcesBuilder {
         }
     }
 
+    private func diagnoseMissingResources(in resources: [Resource]) {
+        // Stop early if the valid resources array count is the same as the user defined list
+        if resources.count == self.target.resources.count {
+            return
+        }
+        
+        // Filter the package list of resources down to those which are not included in the valid resources array
+        let invalidResources = self.target.resources.filter { targetResource in
+            return resources.contains(where: { resource in
+                resource.path.relative(to: self.targetPath).pathString.hasPrefix(targetResource.path)
```

Comment:
Doing a string string prefix comparison here might lead to false positives where the name of a directory entry happens to be the literal prefix of another entry.  That problem seems to exist in other places in `TargetSourcesBuilder` as well, so fixing that across-the-board should perhaps be a separate task.